### PR TITLE
Edit schedule UI improvements

### PR DESCRIPTION
**Pull request by viroulep**
A few small improvements:
  - change fullcalendar's background opacity to have a nice border effect on events
  - fix a sizing bug in the activitypicker (that occurs when we resize the viewport as the picker is collapsed)
  - add a color to a room (that *does* differ from the WCIF spec, if you're fine with it I'll update the spec too).

### CODE REVIEW

File: WcaOnRails/db/migrate/20180629112054_add_color_to_venue_room.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddColorToVenueRoom < ActiveRecord::Migration[5.2]
+  def change
+    add_column :venue_rooms, :color, :string, default: VenueRoom::DEFAULT_ROOM_COLOR, allow_blank: false, limit: 7
```
Comment (jfly) [inline WcaOnRails/db/migrate/20180629112054_add_color_to_venue_room.rb]: What does this `allow_blank` option do? Are you sure you don't mean `null: false` here instead?

I'm not in love with the default being here. I'm concerned that if someone ever changes `DEFAULT_ROOM_COLOR` they won't realize a database migration is needed. Maybe you could instead allow for room's colors to be unspecified, and then the ruby/javascript code could explicitly handle that by falling back to a particular color?

Comment (viroulep) [reply WcaOnRails/db/migrate/20180629112054_add_color_to_venue_room.rb]: I totally meant `null: false`...

Right I'll add an update to all rooms without color to the migration :)
For the new rooms it's already handled through a `before_validation`.

**Comment by viroulep** [comment]
>I think it's useful to have the concept of stage colors! One note is that at the recent us regional competitions that have been going on, stages have been named after the various states in the competition's region.

Isn't this already handled by room names?

File: WcaOnRails/app/javascript/edit-schedule/SchedulesEditor/index.jsx [hunk 2]
```diff
@@ -38,13 +39,21 @@ export class SchedulesEditor extends React.Component {
     };
   }
 
+  componentDidMount() {
+    // We cannot handle well changes (such as room color) when fullCalendar's element is hidden.
+    // So we unselect the room when our panel becomes hidden, to avoid running into any visual bug.
+    $(schedulesEditPanelSelector).find('.panel-collapse').on('hidden.bs.collapse', this.resetSelectedRoom);
```
Comment (viroulep) [inline WcaOnRails/app/javascript/edit-schedule/SchedulesEditor/index.jsx]: This is something I added after reviews.
I noticed that opening a room's schedule then change its color wouldn't update the currently shown calendar (ie: it keeps the old color).
If we keep the color in the props/state, and therefore update the fullcalendar when the color has changed, then we run into a pretty serious issue: fullcalendar's element is not visible at the time of the change, and calling a change of option on it makes it render with its current height: 0... (making it effectively invisible)

The workaround I found was to reset the selected room when the editor's panel is hidden.

File: WcaOnRails/app/javascript/edit-schedule/EditVenue/EditRoom.jsx [hunk 3]
```diff
@@ -3,22 +3,47 @@ import { rootRender } from 'edit-schedule'
...
-      <div className="row">
+      <div className="row room-row">
         <div className="col-xs-9">
           <input type="text" className="form-control" value={roomWcif.name} onChange={this.handleNameChange} />
         </div>
         <div className="col-xs-3">
           <a href="#" onClick={removeRoomAction} className="btn btn-danger pull-right"><i className="fa fa-trash"></i></a>
         </div>
+        <div className="col-xs-9 room-color-cell">
+          <input type="color" className="form-control" value={this.state.color} onChange={this.handleColorChange} onBlur={this.updateColorInWcif} />
```
Comment (viroulep) [inline WcaOnRails/app/javascript/edit-schedule/EditVenue/EditRoom.jsx]: This changed also from the version submitted for review: before we were updating the WCIF whenever the color changed, which can be *a lot* if you drag your cursor around the picker!
(indeed in vanilla javascript the `onchange` event is fired after focus is lost, but in react it's [after any change occurred](https://github.com/facebook/react/issues/6308))

So I decided to keep the color in the state, and only update the WCIF when focus is lost.

**Comment by viroulep** [comment]
I'll merge this in 24h unless someone asks for change ;)

(I commented on the two things I introduced after this PR was approved, but I think they're fine)

File: WcaOnRails/app/models/venue_room.rb [hunk 4]
```diff
@@ -1,19 +1,28 @@
...
   belongs_to :competition_venue
   has_one :competition, through: :competition_venue
   delegate :start_time, to: :competition
   delegate :end_time, to: :competition
   has_many :schedule_activities, as: :holder, dependent: :destroy
 
+  validates :color, format: { with: /\A#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})\z/, message: "Please input a valid hexadecimal color code" }
+
+  before_validation do
+    self.color = DEFAULT_ROOM_COLOR if self.color.blank?
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/venue_room.rb]: Nit: more idiomatically it would be `self.color ||= DEFAULT_ROOM_COLOR` =)

**Comment by viroulep** [comment]
I updated the WCIF spec with this new color attribute.

**Pull request merged**